xdsl/dialects/bufferization.py

```diff
@@ -83,6 +86,11 @@ def verify(self, attr: Attribute, constraint_context: ConstraintContext) -> None
             )
         return self.memref_constraint.verify(memref_type, constraint_context)
 
+    def mapping_type_vars(
+        self, type_var_mapping: dict[TypeVar, AttrConstraint]
+    ) -> Self:
+        return type(self)(self.memref_constraint.mapping_type_vars(type_var_mapping))
```

Comment:
Why this `type(self)`? Do you intend to subclass this constraint?

Reply:
Rewritten